cmd/podman/machine/platform_windows.go

```diff
@@ -1,19 +1,40 @@
 package machine
 
 import (
+	"fmt"
 	"os"
 
+	"github.com/containers/common/pkg/config"
 	"github.com/containers/podman/v4/pkg/machine"
 	"github.com/containers/podman/v4/pkg/machine/hyperv"
 	"github.com/containers/podman/v4/pkg/machine/wsl"
 )
 
-func GetSystemDefaultProvider() machine.VirtProvider {
-	// This is a work-around for default provider on windows while
-	// hyperv is one developer.
-	// TODO this needs to be changed back
-	if _, exists := os.LookupEnv("HYPERV"); exists {
-		return hyperv.GetVirtualizationProvider()
+func GetSystemProvider(vmType *machine.VMType) (machine.VirtProvider, error) {
+	resolvedVMType := machine.QemuVirt
+	if vmType != nil {
+		resolvedVMType = *vmType
+	} else {
```

Comment:
in the same thought here, my understanding is that the containers.conf file would dictate the vm type.  is that what you remember @Luap99 ?

Reply:
If the config file is the only source of truth for this setting, this can be simplified.

Reply:
that was the decision that was made .. i had hoped to also honor environment variables for development and testing.  VMTYPE="qemu" ... wwyt about that too?

Reply:
There is EnvVar support implemented as part of this PR https://github.com/containers/podman/pull/17639/files#diff-1f6113c3d215be90ffaa1c5cc73b7da30d7c6583c49cbae89c4132858478ebf6R25 So, there is an option for development purposes. I just wasn't aware if every configuration should have matching command line parameter override or not.

I will remove extra function from the PR and resubmit.

Reply:
Parameter removed and related branching eliminated from the function.